### DISCUSSION TITLE
ESP32 recommendation with powerful Bluetooth

### DISCUSSION THREAD

**Discussion by blueJack92** [fsaris/EspHome-AwoX-BLE-mesh-hub#109] [General]
I am looking for a powerful, small ESP32 whose Bluetooth performance is particularly good. Unfortunately, my S3s are so weak ... The light hangs 1 meter away and has an RSSI value of 75 or 80.

A light hangs 3 m away and the connection is constantly lost.

Do you have a recommendation for me?

Thank you!

**Comment by fsaris** [comment]
I'm using the WROOM-32 (https://nl.aliexpress.com/item/1005002410521023.html) without any big issues.

A RSSI value between -75 and -80 isn't that bad.  My `min_rssi` is set to `-80`.

is the connection lost or does the ESP32 reboot due to memory issues? Disable logging or set it to `INFO` as logging can have some memory leaks

these are the settings that I'm using to see if it reboots and monitor the free memory.

```
# Enable logging
logger:
  # baud_rate: 9600
  # level: VERBOSE
  logs:
    mdns: INFO
    ota: INFO
    json: INFO
    # component: DEBUG
    mqtt.sensor: INFO
    sensor: WARN
  #   esp32_ble: DEBUG
  #   esp32_ble_client: DEBUG
  #   esp32_ble_tracker: DEBUG
  #   awox.mesh.mqtt: DEBUG
  #   awox.mesh: DEBUG
  #   awox.connection: DEBUG

debug:
  update_interval: 5s

text_sensor:
  - platform: debug
    reset_reason:
      name: "Reset Reason"

sensor:
  - platform: debug
    free:
      name: "Heap Free"
    loop_time:
      name: "Loop Time"
```
